wire-ios-data-model/Source/Model/Conversation/ZMConversation+Predicates.swift

```diff
@@ -77,102 +77,124 @@ extension ZMConversation {
+    public class func predicateForTeamOneToOneConversation() -> NSPredicate {
+        // We consider a conversation being an existing 1:1 team conversation in case the following point are true:
+        //  1. It is a conversation inside a team
+        //  2. The only participants are the current user and the selected user
+        //  3. It does not have a custom display name
 
-        return NSCompoundPredicate(andPredicateWithSubpredicates: [basePredicate, pendingConversationPredicate])
+        let isTeamConversation = NSPredicate(format: "team != NULL")
+        let isGroupConversation = NSPredicate(format: "\(ZMConversationConversationTypeKey) == \(ZMConversationType.group.rawValue)")
+        let hasNoUserDefinedName = NSPredicate(format: "\(ZMConversationUserDefinedNameKey) == NULL")
+        let hasOnlyOneParticipant = NSPredicate(format: "\(ZMConversationParticipantRolesKey).@count == 2")
+
+        return NSCompoundPredicate(andPredicateWithSubpredicates: [isTeamConversation, isGroupConversation, hasNoUserDefinedName, hasOnlyOneParticipant])
     }
 
-    @objc(predicateForClearedConversations)
-    class func predicateForClearedConversations() -> NSPredicate {
-        let cleared = NSPredicate(format: "\(ZMConversationClearedTimeStampKey) != NULL AND \(ZMConversationIsArchivedKey) == YES")
+    class func predicateForConversationsNeedingToBeCalculatedUnreadMessages() -> NSPredicate {
+         return NSPredicate(format: "%K == YES", ZMConversationNeedsToCalculateUnreadMessagesKey)
+    }
 
-        return NSCompoundPredicate(andPredicateWithSubpredicates: [cleared, predicateForValidConversations()])
+    public static func predicateForConversationsArePendingToRefreshMetadata() -> NSPredicate {
+        return NSPredicate(format: "\(ZMConversationIsPendingMetadataRefreshKey) == YES")
     }
 
-    @objc(predicateForConversationsIncludingArchived)
-    class func predicateForConversationsIncludingArchived() -> NSPredicate {
-        let notClearedTimestamp = NSPredicate(format: "\(ZMConversationClearedTimeStampKey) == NULL OR \(ZMConversationLastServerTimeStampKey) > \(ZMConversationClearedTimeStampKey) OR (\(ZMConversationLastServerTimeStampKey) == \(ZMConversationClearedTimeStampKey) AND \(ZMConversationIsArchivedKey) == NO)")
+}
 
-        return NSCompoundPredicate(andPredicateWithSubpredicates: [notClearedTimestamp, predicateForValidConversations()])
-    }
+extension String {
 
-    @objc(predicateForGroupConversations)
-    class func predicateForGroupConversations() -> NSPredicate {
-        let groupConversationPredicate = NSPredicate(format: "\(ZMConversationConversationTypeKey) == \(ZMConversationType.group.rawValue)")
-        let notInFolderPredicate = NSCompoundPredicate(notPredicateWithSubpredicate: predicateForConversationsInFolders())
+    var words: [String] {
```

Comment:
Is it worth to create an extension on string, that is available in all places for all strings? Or can it be a normal `func extractWords(from string: String)` in this class?

If `.words` should be extension on `String`, it would be better searchable and also easier to move to other modules in future as dedicated file.

Reply:
In this PR I just moved existing code into a the factory, refraining from refactoring also code that was moved. I'd like to defer that to another PR.